_posts/2021-09-21-release-1.14.0.md

```diff
@@ -0,0 +1,274 @@
+that use those resources. Through *'slot sharing groups'*, users can influence how the tasks are put
+into slots.
+
+With fine-grained resource management, TaskManager slots are now dynamically sized. Transformations and Operators
+specify what resource profiles they would like, and Flink's Resource Manager and TaskManagers slice
+off that specific part of a TaskManagers total resources. Think of it as a minimal lightweight resource
+orchestration layer within Flink. Hmmmm, why would we even do that, is this not better handled directly by
+Kubernetes or Yarn? For most cases, that is in fact true. But we have seen that in large shared clusters,
+resource requirements change so fast (especially in SQL jobs where each operator needs different resources),
+that a lot of efficiency gets lost waiting for Yarn and K8s to fullfill the resource requests (taking seconds,
+where Flink internally does it in milliseconds).
+
+This feature is mainly useful for large platform deployments where Flink runs many batch
+jobs of varying kind and manages the resources actively (dynamically requesting and releasing resources
+from Kubernetes or Yarn, as opposted to opaque reactive deployments). Please refer to the
+[Fine-grained Resource Management Docs](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/finegrained_resource/)
+for details on how to use this feature.
+
+<figure style="align-content: center">
+  <img src="{{ site.baseurl }}/img/blog/2021-09-25-release-1.14.0/fine_grained_resource_management.png" style="display: block; margin-left: auto; margin-right: auto; width: 600px"/>
+</figure>
+
+Alibaba's internal SQL platform on top of Flink and has used this mechanism for some years now and
+has seen significantly increase the resource utilization of the cluster. This feature was open sourced
+to help other users that build large platforms on top of Flink to optimize resource footprints.
+
+
+# Connectors
+
+## Connector metrics
+
+Flink 1.14 standardized the metrics for connectors ([FLIP-33](https://cwiki.apache.org/confluence/display/FLINK/FLIP-33%3A+Standardize+Connector+Metrics))
+The community will gradually pull them through all connectors, as we rework them
+onto the new unified APIs over the next releases. In Flink 1.14, we cover the Kafka connector,
+and partially the File connectors.
+
+Connectors are the entry and the exit point for data in Flink job. If a job is not running as
+expected, the connector telemetry is among first parts to check. We believe this will become
+a nice improvement for everyone operating production Flink applications.
+
+## Pulsar Source
+
+Flink added a [new Source and Sink](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/connectors/datastream/pulsar/).
+for [Apache Pulsar](https://pulsar.apache.org/). Pulsar is a distributed messaging and streaming
+system.
+
+The connector currently supports the DataStream API. Table API/SQL bindings are expected to be
+contributed next.
```

Comment:
I've rewritten this section to explain in a little detail the features of Pulsar. I hope it will be accepted.

```
In this release, Flink added the [Apache Pulsar](https://pulsar.apache.org/) connector. The Pulsar connector reads data from Pulsar topics in both the streaming and the batch running modes. With the support of the transaction functionality (introduced in Pulsar 2.8.0), the Pulsar source connector provides exactly-once delivery semantic to ensure that a message is delivered exactly once to the consumer, even if a producer retries sending the message. 

And, to ensure message ordering and scaling and serve different purposes, the Pulsar source connector also supports four subscription types:

* [Exclusive](https://pulsar.apache.org/docs/en/concepts-messaging/#exclusive)
* [Shared](https://pulsar.apache.org/docs/en/concepts-messaging/#shared)
* [Failover](https://pulsar.apache.org/docs/en/concepts-messaging/#failover)
* [Key-Shared](https://pulsar.apache.org/docs/en/concepts-messaging/#key_shared)

However, metrics and API unification have not been the only thing that has been done when it comes to connectors. The Apache community will stress improving the connector system. In the future, they will fully implement the Pulsar Sink and the Table API/SQL bindings.

For details about how to use the Pulsar connector, see [Apache Pulsar Connector](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/connectors/datastream/pulsar/#apache-pulsar-connector).
```

Reply:
Thanks, I am adding this to the blog post.